lib/qos/qos.c

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <sys/slist.h>
+#include <qos.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(qos, CONFIG_QOS_LOG_LEVEL);
+
+/* Define a custom STATIC macro that exposes internal variables when unit testing. */
+#if defined(CONFIG_UNITY)
```

Comment:
The test could just include this .c file to access static variables/functions. This would need some changes in the test configuration though.